frontend/src/pages/user-profile/user-profile.tsx

```diff
@@ -0,0 +1,42 @@
+import { useSearchParams } from 'react-router-dom';
+
+import { BackwardButton } from '#libs/components/components.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback } from '#libs/hooks/hooks.js';
+
+import { UserProfileSidebar } from './components/components.js';
+import styles from './styles.module.scss';
+
+const UserProfile: React.FC = () => {
+  const [isSidebarShown, setIsSidebarShown] = useSearchParams({
+    isSidebarShownParam: 'true',
+  });
+
+  const handleButtonBackward = useCallback(() => {
+    setIsSidebarShown({ isSidebarShownParam: 'true' });
+  }, [setIsSidebarShown]);
+
+  const isSidebarShownParameter =
+    isSidebarShown.get('isSidebarShownParam') === 'true' ||
```

Comment:
Let's move it to app.slice. The idea of useSearchParams would be better for selected items for example

Reply:
@v0ldemar01 but @Liza-Veis said that i shouldn`t update or change slices

Reply:
We discussed it before writing a comment and decided to move it to app.slice

Reply:
Attach a link any discussion with coaches for future please

Reply:
it was in the mitting in saturday. timecode: 5:00

Reply:
Good point, sorry for the misclarification, let's stay with search params. But please exclude boolean values from there. The main reason of javascript type coercion. Let's proceed with modes, so we will define some values and will check it

Reply:
done